code/game/objects/items/tools/medical_tools.dm

```diff
@@ -0,0 +1,245 @@
+	to_chat(user, "<span class='warning'>You refill the [src]!</span>")
+	update_icon()
+
+//define for its TOOLs
+/obj/item/tool/geltool/gadget
+	name = "Medical gel based tool"
+	desc = "You shouldn't be holding this! give to high command and let them know how you got it."
+	icon_state = "synthkitoff"
+	item_state = "synthkitoff"
+	flags_atom = CONDUCT
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/items/geltools.dmi'
+	attack_speed = 11 //same as surgery tools for convenience, yet we SHOULDN't need to worry about people beating others with it
+	materials = list(/datum/material/metal = 5000, /datum/material/glass = 2500)//i mean scrap if you want. but thats gonna be a waste for sure
+	var/gelefficiency = 28  //how much gel it uses per application, 28 here will provide 30 uses by default
+	var/obj/item/tool/geltool/gell/cell //Starts with a gell cell
+
+/obj/item/tool/geltool/gadget/Initialize()
+	. = ..()
+	cell = new /obj/item/tool/geltool/gell()
+
+/obj/item/tool/geltool/gadget/update_icon(mob/user,silent=FALSE)
+	..()
+	if(!cell || cell.gelquantity <= 0)
+		item_state = "synthkitoff"
+		playsound(loc, 'sound/weapons/guns/misc/empty_alarm.ogg', 25, 1)
+		to_chat(user, "<span class='warning'>The tool beeps as it runs out of gel!</span>")
+		return
+
+/obj/item/tool/geltool/gadget/proc/gel_use(M = 1)  //change the M if you want to further modify the efficient of a specific procedure
+	cell.gelquantity -= gelefficiency * M
+	usr.visible_message(,"<span class='notice'>the [src] used up [gelefficiency * M] gel in this procedure [cell.gelquantity] left.</span>")
+
+/obj/item/tool/geltool/gadget/attack_hand(mob/living/user)
+	if(user.get_inactive_held_item() != src)
+		return ..()
+	if(!cell)
+		return ..()
+	cell.update_icon()
+	user.put_in_active_hand(cell)
+	cell = null
+	playsound(user, 'sound/machines/click.ogg', 25, 1, 5)
+	playsound(user, 'sound/weapons/guns/interact/flamethrower_unload.ogg', 12, 2, 5)
+	to_chat(user, "<span class='notice'>You remove the Gell from [src].</span>")
+	update_icon(user)
+	return TRUE
+
+/obj/item/tool/geltool/gadget/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(istype(I, /obj/item/tool/geltool/gell))
+		if(!user.drop_held_item())
+			return
+		I.forceMove(src)
+		var/replace_install = "You replace the gell in [src]"
+		if(!cell)
+			replace_install = "You install a gell in [src]"
+		else
+			cell.update_icon()
+			user.put_in_hands(cell)
+		cell = I
+		to_chat(user, "<span class='notice'>[replace_install] <b>Gel quantity: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		playsound(user, 'sound/weapons/guns/interact/rifle_reload.ogg', 25, 1, 5)
+		playsound(user, 'sound/weapons/guns/interact/flamethrower_unload.ogg', 12, 2, 5)
+		update_icon()
+
+// TOOLS down here
+/obj/item/tool/geltool/gadget/kitprinter
+	name = "Synthetic trauma kit"
+	desc = "A tool that utilizes gel. to act as brute or a burn advanced kit. it have a level to toggle its mode"
```

Comment:
```suggestion
	desc = "A tool that utilizes gel to act as brute or a burn advanced kit. It has a button to toggle between modes."
```

Reply:
its actually a lever. in the sprite too. i see the "level" thing but lemme type it right